Tests/Engine/InvokeScriptAnalyzer.tests.ps1

```diff
@@ -611,34 +619,34 @@ Describe "Test -EnableExit Switch" {
     }
 
     # using statements are only supported in v5+
-    if (!$testingLibraryUsage -and ($PSVersionTable.PSVersion -ge [Version]'5.0.0')) {
-        Describe "Handles parse errors due to unknown types" {
+    Describe "Handles parse errors due to unknown types" -Skip:($testingLibraryUsage -or ($PSVersionTable.PSVersion -lt [Version]'5.0.0')) {
+        BeforeAll {
             $script = @'
                 using namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels
                 using namespace Microsoft.Azure.Commands.Common.Authentication.Abstractions
                 Import-Module "AzureRm"
                 class MyClass { [IStorageContext]$StorageContext } # This will result in a parser error due to [IStorageContext] type that comes from the using statement but is not known at parse time
 '@
-            It "does not throw and detect one expected warning after the parse error has occured when using -ScriptDefintion parameter set" {
-                $warnings = Invoke-ScriptAnalyzer -ScriptDefinition $script
-                $warnings.Count | Should -Be 1
-                $warnings.RuleName | Should -Be 'TypeNotFound'
-            }
+        }
+        It "does not throw and detect one expected warning after the parse error has occured when using -ScriptDefintion parameter set" {
+            $warnings = Invoke-ScriptAnalyzer -ScriptDefinition $script
+            $warnings.Count | Should -Be 1
+            $warnings.RuleName | Should -Be 'TypeNotFound'
+        }
 
+        It "does not throw and detect one expected warning after the parse error has occured when using -Path parameter set" {
             $testFilePath = "TestDrive:\testfile.ps1"
-            Set-Content $testFilePath -value $script
-            It "does not throw and detect one expected warning after the parse error has occured when using -Path parameter set" {
-                $warnings = Invoke-ScriptAnalyzer -Path $testFilePath
-                $warnings.Count | Should -Be 1
-                $warnings.RuleName | Should -Be 'TypeNotFound'
-            }
+            Set-Content $testFilePath -Value $script
+            $warnings = Invoke-ScriptAnalyzer -Path $testFilePath
+            $warnings.Count | Should -Be 1
+            $warnings.RuleName | Should -Be 'TypeNotFound'
         }
+    }
 
-        Describe "Handles static Singleton (issue 1182)" {
-            It "Does not throw or return diagnostic record" {
-                $scriptDefinition = 'class T { static [T]$i }; function foo { [CmdletBinding()] param () $script:T.WriteLog() }'
-                Invoke-ScriptAnalyzer -ScriptDefinition $scriptDefinition -ErrorAction Stop | Should -BeNullOrEmpty
-            }
+    Describe "Handles static Singleton (issue 1182)"-Skip:($testingLibraryUsage -or ($PSVersionTable.PSVersion -lt [Version]'5.0.0')) {
```

Comment:
```suggestion
    Describe "Handles static Singleton (issue 1182)" -Skip:($testingLibraryUsage -or ($PSVersionTable.PSVersion -lt [Version]'5.0.0')) {
```

Reply:
Good spot, fixed in 5ee8a380bf07ade4532a98e44421734e47b2f913